src/Reflection/ReflectionClass.php

```diff
@@ -90,6 +90,8 @@ class ReflectionClass implements Reflection
 
     private ?ReflectionClass $cachedParentClass = null;
 
+    private static ReflectionClass|false|null $cachedStringableInterface = null;
```

Comment:
A `static` is not viable here, because in projects like `roave/backwards-compatibility-check`, different/independent reflectirs are used.

I don't have a solution for this: perhaps taking the performance impact, and optimizing reflector hit/miss operations (overall) could be a better way? 🤔

Reply:
@Ocramius 

I've just found out that we already have `MemoizingSourceLocator`: https://github.com/Roave/BetterReflection/blob/5.0.x/src/SourceLocator/Type/MemoizingSourceLocator.php#L16

And it's default source locator for default reflector: https://github.com/Roave/BetterReflection/blob/5.0.x/src/BetterReflection.php#L46